File: lib/files.com/models/image.rb
# frozen_string_literal: true

module Files
  class Image
    attr_reader :options, :attributes

    def initialize(attributes = {}, options = {})
      @attributes = attributes || {}
      @options = options || {}
    end

    # string - Image name
    def name
      @attributes[:name]
    end

    # string - Image URI
    def uri
      @attributes[:uri]
    end
  end
end

File: dotnet/sdk/Files.com/Models/Image.cs

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Files.com.Models
{
    public class Image
    {
        private Dictionary<string, object> attributes;
        private Dictionary<string, object> options;

        public Image()
        {
            this.attributes = new Dictionary<string, object>();
            this.options = new Dictionary<string, object>();

            this.attributes.Add("name", null);
            this.attributes.Add("uri", null);
        }

        public Image(Dictionary<string, object> attributes, Dictionary<string, object> options)
        {
            this.attributes = attributes;
            this.options = options;
        }

        /// <summary>
        /// Image name
        /// </summary>
        [JsonPropertyName("name")]
        public string Name
        {
            get { return (string) attributes["name"]; }
        }

        /// <summary>
        /// Image URI
        /// </summary>
        [JsonPropertyName("uri")]
        public string Uri
        {
            get { return (string) attributes["uri"]; }
        }



    }
}

